src/lib/utils/tracing/probes.hpp

```diff
@@ -57,4 +57,25 @@ constexpr bool is_valid_name(const char* name) {
                        param10, param11, param12)                                                                    \
   BUILD_PROBE_NAME(provider, probe, param1, param2, param3, param4, param5, param6, param7, param8, param9, param10, \
                    param11, param12);
+
+#else
+
+// Because TSan has issues and throw false positives, we don't use probes in TSan builds
+#ifdef __SANITIZE_THREAD__
+
+#undef DTRACE_PROBE
```

Comment:
Wait, why does this even work? Shouldn't this lead to an error if the macro is not defined at all?

Reply:
Since it did not lead to an error: it must be redefined somewhere else?

Reply:
Maybe check what happens if you remove the ifdef?

Reply:
Then, everything fails. Apparently, `#ifdef __SANITIZE_THREAD__` does not work properly otherwise the TSAN test on the CI should have failed.

In addition, undefing all the DTRACE_PROBEs is not an option. Assuming that we get `#ifdef __SANITIZE_THREAD__` to work, we could guard all occurrences of probes, but this would clutter the code and I would hate to do that.

Reply:
Simply redef it as something that doesn't do anything.

Maybe using feature macros works? https://clang.llvm.org/docs/ThreadSanitizer.html

Reply:
Right, we already use that one for the SingletonTest.

Reply:
There is even `__attribute__((no_sanitize("thread")))` as a function attribute, interesting. I will go with the feature macro anyway.